src/hooks/index.ts

```diff
@@ -48,7 +48,6 @@ export {
   UserDomainReputation,
 } from './useUserReputationForTopDomains';
 export { default as useColonyContext } from './useColonyContext';
-export { default as useUserByNameOrAddress } from './useUserByNameOrAddress';
```

Comment:
I don't think there was a use case for this hook (or at least not one strong enough to justify its overheads) and I felt like it was misused a little.

Under the hood, it would call 2 queries (get user by address and by name), and then returned the results of one of them based on the arguments provided (which was already quite wasteful since we could tell if the argument is an address or username before querying).

Analysing all its usage, the majority of cases called only for the get user by address query rather than name, so I went ahead and replaced those with a hook doing specifically that. There was one case where it was truly getting the user by either name or address (the parameter in user profile route was accepting either one or another) but I've checked in the Dapp and we didn't allow the address in the profile route there, so I made CDapp consistent with that. Let me know if this was a new requirement and should be kept, and I'll go back and revert.

Reply:
The question about the user profile route param would be one for @rdig